Sources/Xcodeproj/Plist.swift

```diff
@@ -0,0 +1,50 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// A enum representing data types for legacy Plist type.
+/// see: https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.html
+enum Plist {
+    case string(String)
+    case array([Plist])
+    case dictionary([String: Plist])
+}
+
+extension Plist: ExpressibleByStringLiteral {
+    public typealias UnicodeScalarLiteralType = StringLiteralType
+    public typealias ExtendedGraphemeClusterLiteralType = StringLiteralType
+
+    public init(unicodeScalarLiteral value: UnicodeScalarLiteralType) {
+        self = .string(value)
+    }
+    public init(extendedGraphemeClusterLiteral value: ExtendedGraphemeClusterLiteralType) {
+        self = .string(value)
+    }
+    public init(stringLiteral value: StringLiteralType) {
+        self = .string(value)
+    }
+}
+
+private let escapedQuote = "\\\""
+
+extension Plist {
+    /// Serializes the Plist enum to string.
+    func serialize() -> String {
+        switch self {
+        case .string(let str):
+            // If the string has spaces Xcode represents it as: `"\"some string\""`.
+            let hasSpace = str.utf8.contains(UInt8(ascii: " "))
+            return "\"" + (hasSpace ? escapedQuote : "") + str.escape(ascii: "\"") + (hasSpace ? escapedQuote : "") + "\""
```

Comment:
This is actually not a core part of the plist specification.  What you sometimes see in Xcode is the result of transformations at various layers:  if a component of a build setting that is a string array contains whitespace, then that component gets quotes around it and any quotes and escapes inside of it get escaped.  Then, when serialized to an OPENSTEP-style plist, another layer of quotes and escapes are used to represent those in the content.

So in this case, a general-purpose plist (as this is) should just quote the value (and escape any quotes or escapes) within it.  Whether a particular build setting is quoted should happen at the caller layer.